packages/enhanced/src/lib/container/ContainerPlugin.ts

```diff
@@ -210,35 +217,98 @@ class ContainerPlugin {
           },
           (error: WebpackError | null | undefined) => {
             if (error) return callback(error);
-            if (hasSingleRuntimeChunk) {
-              // Add to single runtime chunk as well.
-              // Allows for singleton runtime graph with all needed runtime modules for federation
-              addEntryToSingleRuntimeChunk();
-            } else {
-              callback();
-            }
+            hooks.addContainerEntryModule.call(dep);
+            callback();
           },
         );
+      },
+    );
+
+    compiler.hooks.finishMake.tapAsync(
```

Comment:
once make hook has run, we should have all entrypoints in compilation - then i can check if there are multiple runtime chunks. If there are multiple runtime chunks, create a containerEntry for each runtime so there is a container modules thats coupled to each possible runtime